### PR TITLE
Topics pagination

### PR DESCRIPTION
**Pull request by Mentessi**
adding a new spec file for pagination felt right as they'll be lots more pagination to come - collections, users, search results ... but wasn't 100% sure? 

Other that this, it seemed like a fairly straightforward feature, this could well be a sign I've missed something :D

### CODE REVIEW

File: spec/factories/topics.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 FactoryBot.define do
   factory :topic do
-    name "What is your secret superpower?"
+    sequence :name do |n|
+      "This is my topic number # #{n}"
```
Comment (Neilos) [inline spec/factories/topics.rb]: So there's a subtle error that can occur with sequences.
Be aware that when you have lots of specs you can easily end up with tens and even hundreds of instances being generated with a factory.

Consider the following expectation:
```ruby
expect(page).to have_text("This is my topic number 2")
```

This will pass for all of the following topics:
```
  "This is my topic number 2"
  "This is my topic number 20"
  "This is my topic number 21"
  "This is my topic number 22"
  ...
  "This is my topic number 200"
  "This is my topic number 201"
  "This is my topic number 202"
  ...
  "This is my topic number 220"
  ...
  etc
```

To avoid this problem, it is best to produce zero padded numbers.
For example:
```
  "This is my topic number 002"
  "This is my topic number 021"
  "This is my topic number 221"
```

You can do this with Ruby string formatting (`%`). See Ruby string object. This uses the `Kernel::sprintf` formatting options, so have a look at that too.

File: spec/system/managing_topics_spec.rb [hunk 2]
```diff
@@ -4,17 +4,29 @@
 
 RSpec.describe 'managing topics' do
   let(:user) { FactoryBot.create :user }
-  let!(:topic) do
+  let!(:topic) {
     FactoryBot.create :topic, name: 'What is your favourite colour'
-  end
+  }
+
+  let!(:topics){ FactoryBot.create_list(:topic, 31) }
```
[the remark this answers is not in the document]
Comment (Mentessi) [reply spec/system/managing_topics_spec.rb]: Makes sense - I was overwriting the WillPaginate defaults in the topics controller so I have now just moved this to the config with different settings for development and production... I'm not sure I've done this right, referencing them using Rails.configuration.x.WillPaginate.per_page isn't something I've seen before, but it was the only way I could get it working

File: config/environments/production.rb [hunk 3]
```diff
@@ -82,6 +82,8 @@
     authentication: :plain,
   }
 
+  config.x.WillPaginate.per_page = 15
```
Comment (Neilos) [inline config/environments/production.rb]: Like the x 👍
Any reason you departed from snake case for the WillPaginate? (i.e why not `will_paginate`?)

Or even just `config.x.results_per_page`? The variable name should be whatever is meaningful to your app and your code, and it shouldn't assume the usage. What if you later decide you need your own code (i.e. not just will_paginate gem) to access this config value for some reason or what if you chose to change the pagination gem you're using? Make the name descriptive, but not specific to particular code implementations.

File: spec/factories/topics.rb [hunk 4]
```diff
@@ -2,7 +2,9 @@
 
 FactoryBot.define do
   factory :topic do
-    name 'What is your secret superpower?'
+    sequence :name do |n|
+      "This is my topic number # #{"%.8d" % n}"
```
Comment (Neilos) [inline spec/factories/topics.rb]: This is a useful change, but not really part of this pull request.

File: config/environments/production.rb (hunk 3 above)
Comment (Mentessi) [reply config/environments/production.rb]: I followed the readme for will paginate...
```
# set per_page globally
WillPaginate.per_page = 10
```
I didn't realise that using config.x.variable_name was possible and just setting a variable. 

Will make these changes :)

File: spec/factories/topics.rb (hunk 4 above)
Comment (Mentessi) [reply spec/factories/topics.rb]: this change is because I could get false positives with my specs due to the way I was using sequences before and then generating a large number of topics (this was your original feedback :D - further up the page!) 

Happy to take it out though... ?

Comment (Neilos) [reply spec/factories/topics.rb]: 😳 ok.
